test/e2e/volume_provisioning.go

```diff
@@ -38,15 +45,40 @@ import (
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
+type storageClassTest struct {
+	name           string
+	cloudProviders []string
+	provisioner    string
+	parameters     map[string]string
+	claimSize      string
+	expectedSize   string
+	pvCheck        func(volume *v1.PersistentVolume) error
+}
+
 const (
-	// Requested size of the volume
-	requestedSize = "1500Mi"
 	// Plugin name of the external provisioner
 	externalPluginName = "example.com/nfs"
 )
 
-func testDynamicProvisioning(client clientset.Interface, claim *v1.PersistentVolumeClaim, expectedSize string) {
-	err := framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client, claim.Namespace, claim.Name, framework.Poll, framework.ClaimProvisionTimeout)
+func testDynamicProvisioning(t storageClassTest, client clientset.Interface, claim *v1.PersistentVolumeClaim, class *storage.StorageClass) {
+	if class != nil {
+		By("creating a StorageClass " + class.Name)
+		class, err := client.Storage().StorageClasses().Create(class)
```

Comment:
I am not sure it's mandatory, all controllers use Core(). Anyway, I changed it to V1